mitreattack/collections/collection_to_index.py

```diff
@@ -0,0 +1,174 @@
+import argparse
+import os
+from tqdm import tqdm
+import json
+import uuid
+from dateutil.parser import isoparse
+import re
+
+
+class CollectionToIndex:
+    @staticmethod
+    def generate_index(name, description, root_url, files=None, folders=None, bundles=None):
+        """generate a collection index from the input data and return the index as a dict
+        arguments:
+            name (string): the name of the index
+            description (string): the description of the index
+            root_url (string): the root URL where the collections can be found. Specified collection paths will be
+                                appended to this for the collection URL
+            files (string[], optional): collection JSON files to include in the index. Cannot be used with folder
+                                argument
+            folders (string[], optional): folders of collection JSON files to include in the index. Cannot be used with
+                                files argument. Will only match collections that end with a version number
+            bundles (stix bundle[], optional): array of stix bundle objects to include in the index
```

Comment:
I'm concerned that the word "bundles" (and the documented type "stix bundle[]" will make users think that we are accepting a STIX 2.1 Bundle object[[1](https://stix2.readthedocs.io/en/latest/api/v21/stix2.v21.bundle.html)][[2](https://github.com/oasis-open/cti-python-stix2/blob/master/stix2/v21/bundle.py#L11)] from the STIX2 library, and not simply the plain JSON-dict representation of a bundle.

Reply:
On a similar note, do we want to support users who want to use STIX2.1 MemoryStores as input? Pretty much all of our [documentation for accessing ATT&CK](https://github.com/mitre-attack/attack-stix-data/blob/master/USAGE.md#access-the-most-recent-version-via-memorystore) content utilizes MemoryStores so it might be possible that a user would load in memory stores for a few versions of ATT&CK and then want to build a collection index from them.

We could either:
- Allow bundles to take either a plain array of dicts or an array of MemoryStores (and infer the type based on what it sees)
- Add an additional optional argument for MemoryStore inputs

Reply:
I changed the name to "sets", which I think is reasonably similar but not confusing. I also added support for MemoryStores via the first of the two suggested approaches.